src/Datagrid/SimplePager.php

```diff
@@ -77,6 +77,11 @@ public function getResults($hydrationMode = null)
         if (\count($this->results) > $this->getMaxPerPage()) {
             $this->haveToPaginate = true;
 
+            @trigger_error(
+                sprintf('The %s method will not return %s starting from 4.0', __METHOD__, ArrayCollection::class),
```

Comment:
```suggestion
                sprintf('Returning a type different than array from %s is deprecated since sonata-project/admin-bundle 3.x and will return array starting from 4.0.', __METHOD__),
```

Reply:
👍